internal/pkg/cli/env_package.go

```diff
@@ -0,0 +1,298 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/copilot-cli/internal/pkg/cli/deploy"
+	"github.com/spf13/afero"
+
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+type packageEnvVars struct {
+	envName      string
+	appName      string
+	outputDir    string
+	uploadAssets bool
+}
+
+type discardFile struct{}
+
+func (df discardFile) Write(p []byte) (n int, err error) {
+	return io.Discard.Write(p)
+}
+
+func (df discardFile) Close() error {
+	return nil //noop
+}
+
+type packageEnvOpts struct {
+	packageEnvVars
+
+	// Dependencies.
+	cfgStore     store
+	ws           wsEnvironmentReader
+	sel          wsEnvironmentSelector
+	caller       identityService
+	fs           afero.Fs
+	tplWriter    io.WriteCloser
+	paramsWriter io.WriteCloser
```

Comment:
We can use `WriteCloser` for `svc package`'s writers as well! In a later refactor